File: core-java-practice/leet-code-codebase/twoSum.java

import java.util.Scanner;

public class twoSum {

    public static void main(String[] args) {
        //  Scanner for read input from the user
        Scanner sc = new Scanner(System.in);
        //  read size of an array
        int n = sc.nextInt();
        //  define the array
        int[] arr = new int[n];
        //  read array elements
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        //  read terget sum
        int tar = sc.nextInt();

        // find the indices of two no. that sum is euqal to target
        boolean flag = false;
        for (int i = 0; i < n && !flag; i++) {
            for (int j = i + 1; j < n; j++) {
                if (arr[i] + arr[j] == tar) {
                    System.out.println(i + " " + j);
                    flag = true;
                    break;
                }
            }
        }
    }
}

File: core-csharp-practice/leet-code-codebase/TwoSum.cs
using System;

class TwoSum
{
    static void Main()
    {
        // Read size of array
        Console.WriteLine("Enter size of array:");
        string inputSize = Console.ReadLine();

        if (!int.TryParse(inputSize, out int n) || n <= 0)
        {
            Console.WriteLine("Invalid array size");
            return;
        }

        int[] arr = new int[n];

        // Read array elements
        Console.WriteLine("Enter array elements:");
        for (int i = 0; i < n; i++)
        {
            int.TryParse(Console.ReadLine(), out arr[i]);
        }

        // Read target sum
        Console.WriteLine("Enter target sum:");
        string inputTarget = Console.ReadLine();

        if (!int.TryParse(inputTarget, out int target))
        {
            Console.WriteLine("Invalid target");
            return;
        }

        // Find indices of two numbers whose sum equals target
        bool found = false;

        for (int i = 0; i < n && !found; i++)
        {
            for (int j = i + 1; j < n; j++)
            {
                if (arr[i] + arr[j] == target)
                {
                    Console.WriteLine($"{i} {j}");
                    found = true;
                    break;
                }
            }
        }

        if (!found)
        {
            Console.WriteLine("No pair found");
        }
    }
}
